f21/ps.html

```diff
@@ -858,50 +857,47 @@ <h2><u> Lecture / Lab 6</u></h2>
 <br><b>Note:</b> The output will not display a username on Gradescope  since there is no login on the cloud image where the grading script runs.
 
 
-<li > <b>Due Date: 24 October</b>
+<li > <b>Due Date: 27 October</b>
 <b style="float:right">Reading: <a href="https://pandas.pydata.org/pandas-docs/stable/user_guide/10min.html">10-mins to Pandas</a>, <a href="https://www.datacamp.com/community/tutorials/pandas-tutorial-dataframe-python#gs.hnyRaik">DataCamp Pandas</a> & <a href="lab6.html">Lab 6</a></b><br>
 
 
-<p> Download the <a href="../files/volcanicEruptions.csv">Volcanos dataset</a> and look at the data and, in particular, the column names.
+<p> Download the <a href="../files/ramenRatings.csv">Ramen dataset</a> and look at the data and, in particular, the column names.
 <p> Then, write a program that
 <ul>
 <li> Ask the user for the name of the input file.
-<li> Print the average elevation for each Dominant Rock Type
-<li> Print the height of the tallest volcano in Italy
+<li> Print the average rating, out of 5 stars, per serving style.
+<li> Print the lowest rating for a ramen spot in Singapore, and the name of the place.
 </ul>
 <p>
 <p>
 <pre style ="background-color:lavender;width:500px">
-Enter file name: volcanicEruptions.csv
-
-The average height per rock type is as follows:
-Dominant Rock Type
-Andesite / Basaltic Andesite                2153.494810
-Basalt / Picro-Basalt                       1129.892495
-Dacite                                      2538.904255
-Foidite                                     1519.600000
-No Data                                      969.810345
-Phono-tephrite /  Tephri-phonolite          1390.333333
-Phonolite                                   2784.222222
-Rhyolite                                    1564.363636
-Trachyandesite / Basaltic Trachyandesite    1820.700000
-Trachybasalt / Tephrite Basanite            1506.287671
-Trachyte / Trachydacite                     2070.555556
-Name: Elevation (Meters), dtype: float64
-
-The tallest volcano in Italy is 3330 meters high.
+Enter file name:ramenRatings.csv
+
+The average stars per serving style:
+Style
+Bar     5.000000
+Bowl    3.937165
+Box     4.291667
+Can     3.500000
+Cup     3.619075
+Pack    3.863330
+Tray    3.682203
+Name: Stars, dtype: float32
+
+KOKA has the lowest rating in Singapore with 2.5 stars.
 </pre>
 
-<p><b>Hint: </b>You should use <tt>groupby()</tt> and <tt>get_group()</tt> as described in <a href="lab6.html">Lab 6</a>.
-
+<p>
+<b>Hint: </b>You should use <tt>groupby()</tt> and <tt>get_group()</tt> as described in <a href="lab6.html">Lab 6</a> as well as <tt>idxmin()</tt>.
```

Comment:
since we don't cover idxmin() in class, please add a link to a reference here and add a sentence to what they should use it for. Thanks!

Reply:
Okay, I'll use this: https://www.geeksforgeeks.org/python-pandas-dataframe-idxmin/